subiquity/server/controllers/kernel_crash_dumps.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from typing import TypedDict
+
+from subiquity.server.controller import NonInteractiveController
+
+log = logging.getLogger("subiquity.server.controllers.kernel_crash_dumps")
+
+KernelCrashDumpsConfig = TypedDict(
+    "KernelCrashDumpsConfig",
+    {
+        "enabled": bool,
+    },
+    total=True,
+)
```

Comment:
Alternative notation, (which I like better but it's up to you)

```python
class KernelCrashDumpsConfig(TypedDict, total=True):
    enabled: bool
```

Reply:
Sure, I like this format better too. We'll have to change it back if we eventually expand the config to allow the originally planned "memory-*" keys but we should use the better format where we can!

Reply:
Either way is fine, specially if you anticipate keys with `-`.